src/rules/__tests__/prefer-hooks-on-top.test.ts

```diff
@@ -0,0 +1,200 @@
+import { TSESLint } from '@typescript-eslint/experimental-utils';
+import rule from '../prefer-hooks-on-top';
+
+const ruleTester = new TSESLint.RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+  },
+});
+
+ruleTester.run('basic describe block', rule, {
+  valid: [
+    [
+      'describe("foo", () => {',
+      '  beforeEach(() => {',
+      '  });',
+      '  someSetupFn();',
+      '  afterEach(() => {',
+      '  });',
+      '  test("bar", () => {',
+      '    some_fn();',
+      '  });',
+      '});',
+    ].join('\n'),
+  ],
+  invalid: [
+    {
+      code: [
+        'describe("foo", () => {',
+        '  beforeEach(() => {',
+        '  });',
+        '  test("bar", () => {',
+        '    some_fn();',
+        '  });',
+        '  beforeAll(() => {',
+        '  });',
+        '  test("bar", () => {',
+        '    some_fn();',
+        '  });',
+        '});',
+      ].join('\n'),
+      errors: [
+        {
+          messageId: 'noHookOnTop',
+          column: 3,
```

Comment:
I tried with changing this test to template string. The column changed from `3` to `11` because it had to take into account the extra indentation.

I'm ok with both the approaches.

A question though, for a first time contributor to this project, would it be ok to expect them to know about the extra indent. It sound like a gotcha they would face when writing the test

Reply:
Personally I think it's fine - You can always de-indent the block, and modify it like that, or like I said call trim, then once you know you've gotten it correct, add the indent amount or copy them from the test error.

While still not optimal, I think it's better than `join` as it makes it very speedy to add new test cases.